### PR TITLE
test: include test for FindCommand and its parser

### LINKED ISSUE #67: test: create Test files for FindCommand

### LINKED ISSUE #114: test: create test file for FindCommandParser

### LINKED ISSUE #115: test: create test file for FindCommand

### PR DESCRIPTION
**Pull request by yusufaine**
Overview of changes:
1. FindCommand now supplies an immutable list to RecipeContainsKeywordPredicate
2. FindCommand test should cover most combinations of cases
3. Uncommented CommandParserUtil, no changes were needed and this change was necessary to test FindCommandParser, and I assume all the other *Parser files.
4. Included a `filteredListEquality` method in `TestUtil` as List equality also checks if an item is at the same index as another item.
    - If object A is at index 3 in List A, object A must also be at index 3 of List B for them to be equal. 
5. Minor change in `getTypicalRecipes` to make it immutable.

@AY2122S2-CS2103T-T17-2/developers try to merge this issue first as I feel like you guys may need `CommandParserUtil`.

Closes #114.
Closes #115.

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/FindCommandParserTest.java [hunk 1]
```diff
@@ -1,34 +1,33 @@
...
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+    }
+
+    @Test
+    public void parse_validArgs_returnsFindCommand() throws ParseException {
+        RecipeContainsKeywordPredicate pred = new RecipeContainsKeywordPredicate(List.of("Western", "Bacon"));
+        FindCommand expected = new FindCommand(pred);
+
+        assertParseSuccess(parser, "Western, Bacon", expected);
+        assertParseSuccess(parser, "      Western \r\n\t\t,    Bacon", expected);
```
Comment (aidanaden) [inline src/test/java/seedu/address/logic/parser/FindCommandParserTest.java:31]: Minor: should an input that includes newline breaks be considered valid? am not too sure bout this (it's a v small edge case tho)

**Review by aidanaden — APPROVED** [review]
LGTM! Just a minor consideration regarding validity of inputs with newlines/tabs.

🍝 🍝 🍝

File: src/test/java/seedu/address/logic/parser/FindCommandParserTest.java (hunk 1 above)
Comment (yusufaine) [reply src/test/java/seedu/address/logic/parser/FindCommandParserTest.java:31]: For this part, the Parser would comma-separate the values to "      Western \r\n\t\t", and "    Bacon". The point of that assertion is to ensure that the white spaces of each CSV is being trimmer to prevent any unexpected behaviours.

**Review by blaketingyu — APPROVED** [review]
LGTM !! 🌮

**Review by gigituang — APPROVED** [review]
LGTM! 🌮
